### PR TITLE
MP OpenAPI 4.0 initial implementation

### PR DESCRIPTION
**Pull request by Azquelt**
Deliver the initial implementation of the `mpOpenAPI-4.0` feature.

- Refactor existing implementation to separate out the parts that need to differ in the new implementation
- Add upstream dependencies
  - MP OpenAPI spec API 4.0-RC1
  - SmallRye OpenAPI 4.0.0-alpha1
  - Jandex 3.2
- New implementation code
- Add TCK FAT project

Not covered by this PR:
- Validation of the final OpenAPI document
- Merging of OpenAPI documents from different modules

### CODE REVIEW

File: dev/io.openliberty.microprofile.openapi.4.0.internal.services/src/io/openliberty/microprofile/openapi40/internal/services/impl/OpenAPI31Validator.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import io.openliberty.microprofile.openapi20.internal.services.OASValidationResult;
+import io.openliberty.microprofile.openapi20.internal.services.OASValidator;
+
+@Component(service = OASValidator.class, configurationPolicy = ConfigurationPolicy.IGNORE)
+public class OpenAPI31Validator implements OASValidator {
+
+    @Override
+    public OASValidationResult validate(OpenAPI model) {
+        // TODO: implement validation
```
Comment (abutch3r) [inline dev/io.openliberty.microprofile.openapi.4.0.internal.services/src/io/openliberty/microprofile/openapi40/internal/services/impl/OpenAPI31Validator.java:24]: I assume this is will be addressed in the next phase of work for OAI

Comment (Azquelt) [reply dev/io.openliberty.microprofile.openapi.4.0.internal.services/src/io/openliberty/microprofile/openapi40/internal/services/impl/OpenAPI31Validator.java:24]: Yeah, validation and merging will be done later.

**Review by abutch3r — DISMISSED** [review]
LGTM - have doubled checked where new files had old Copyright dates that they were taken from other files and I am satisfied on that.

**Comment by Azquelt** [comment]
#build
#libby

**Comment by Azquelt** [comment]
Build failed because I hadn't allowed the TCK project to pull from maven central.

#libby
#build

**Comment by Azquelt** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.microprofile.openapi_fat,com.ibm.ws.openapi.ui_fat,io.openliberty.microprofile.openapi.2.0.internal_fat,io.openliberty.microprofile.openapi.ui.internal_fat,com.ibm.ws.microprofile.openapi.1.1_fat_tck,com.ibm.ws.microprofile.openapi_fat_tck,io.openliberty.microprofile.openapi.2.0.internal_fat_tck,io.openliberty.microprofile.openapi.3.0.internal_fat_tck,io.openliberty.microprofile.openapi.3.1.internal_fat_tck,io.openliberty.microprofile.openapi.4.0.internal_fat_tck

**Review by abutch3r — APPROVED** [review]
Assuming tests now build and pass

**Comment by Azquelt** [comment]
Failures in the build are known issues.

**Pull request merged**